npm/src/upload/index.ts

```diff
@@ -43,6 +50,17 @@ export class FileUploader {
   ) => {
     window.addEventListener("beforeunload", pageUnloadAction)
 
+    const intervalId = setInterval(() => {
```

Comment:
Since we know the expiry time of the token, is it possible to use `setTimeout` rather than `setInterval`? Or is there some reason you need to keep checking rather than relying on the original token expiry time?

Reply:
From what I've seen when the access token is refreshed, the refresh token exp is reset. So it needs to be continually checked, I think using `setTimeout` means the reset only occurs once, which won't work where the idle session is more than an hour. I'm not sure what you mean by "... relying on the original token expiry time". The function `idleSessionTimeoutAboutToExpire` is relying on the current refresh token expiry to calculate whether the idle session timeout is going to expire.

Reply:
So is the access token refreshed automatically - we can't add something that runs after every the token refresh?

Reply:
The access token is not refreshed automatically. Can do something like this though:

```
checkIdleSessionTimeOut: () => void = () => {
    //Refresh token will expire after 30mins, therefore check 1min before this, ie in 29 mins
    const timeoutInMs = 29 * 60 * 1000
    setTimeout(() => {
      console.log("Refreshing idle session")
      refreshOrReturnToken(this.keycloak).then(() => this.idleSession())
    }, timeoutInMs)
  }
...
uploadFiles: (
    files: IFileWithPath[],
    uploadFilesInfo: FileUploadInfo
  ) => Promise<void> = async (
    files: IFileWithPath[],
    uploadFilesInfo: FileUploadInfo
  ) => {
    window.addEventListener("beforeunload", pageUnloadAction)
    this.idleSession()
    ...
}
```
Can also calculate the timeout using "now" and the refresh token expiry value, instead of having a fixed value.
Assuming you're wanting avoid a continuous call to check the state of the refresh token?

Reply:
The access token is not refreshed automatically when the session is idle, ie when an upload is occurring.

Is the problem that there is a continual check on the refresh token on page? Could have a function something like this, that would only attempt the refresh just before the refresh token is due to expire:

```
idleSession: () => void = () => {
    const nowInSecs = Math.round(new Date().getTime() / 1000)
    const refreshToken = this.keycloak.refreshTokenParsed
    if (refreshToken != undefined && refreshToken.exp != undefined) {
      const expInSecs = refreshToken.exp
      const timeoutInMs = (expInSecs - (nowInSecs + 60)) * 1000
      console.log("Timeout in Ms: " + timeoutInMs)
      setTimeout(() => {
        console.log("Refreshing idle session")
        refreshOrReturnToken(this.keycloak).then(() => this.idleSession())
      }, timeoutInMs)
    }
  }
```

Reply:
Maybe we could have a quick call about this? It sounds like I don't understand the intended workflow. 🙂